planner/core/find_best_task.go

```diff
@@ -1035,9 +1035,10 @@ func (ds *DataSource) getOriginalPhysicalTableScan(prop *property.PhysicalProper
 		ts.StoreType = kv.TiKV
 	}
 	if ts.StoreType == kv.TiFlash {
+		// append the AccessCondition to filterCondition because TiFlash only support full range scan for each
+		// region, do not reset ts.Ranges as it will help prune regions during `buildCopTasks`
```

Comment:
Since it's a full range scan. Which task is pruned before?

Reply:
For example, for a table t with id as its primary key, and the id is the handle column, if user fire a query like `select count(*) from t where id >= 100`, then obviously, only some of the regions have the useful data. For TiKV, TiDB will only send the cop request to these regions, this "task prune" is done by passing range [100, +inf) to the `buildCopTask` function. For TiFlash, although it requires full range scan on every region, we can still prune  the task which is sending cop request to the regions that do not have data with `id >= 100`